salt/states/zabbix_host.py

```diff
@@ -342,10 +349,17 @@ def _interface_format(interfaces_data):
                         }
                     else:
                         hostintf["details"] = []
-            interface_diff = [
-                x for x in interfaces_formated if x not in hostinterfaces_copy
-            ] + [y for y in hostinterfaces_copy if y not in interfaces_formated]
-            if interface_diff:
+            # cycle through all of the interfaces from both the API and the SLS and keep track of how many "match"
+            # This doesn't look for exact matches, but looks for subsets. This allows more items to come from the API
+            # than from the SLS and we don't need to worry about not being able to set options like 'error'
+            # Once we have the count of how many interfaces match, we compare that to the number of interfaces that
+            # are already on the host. If the number is different, we trigger an update. Otherwise we just let it be
+            matching_interfaces = 0
+            for interface_formated in interfaces_formated:
+                for hostinterface_copy in hostinterfaces_copy:
+                    if hostinterface_copy.items() >= interface_formated.items():
+                        matching_interfaces += 1
+            if matching_interfaces != len(hostinterfaces_copy):
```

Comment:
If the host already have one interface and you try to add a new one, this code will not trigger the update process.
The `len(hostinterfaces_copy)` will be `1`, because the host already have the first interface and the comparision will produce one match so, the check at line 362 will not set the `update_interfaces = True`.

Reply:
Hi @rsolano-isb ,

I wrote a fix for this interface compare problem (removing the read-only fields from Zabbix answers) some time ago, but I was waiting to do more extensive tests before open the PR. Well, just openned the PR #61585 with my fix, even without the extensive tests.

But if you can fix your code to work adding and removing interfaces I guess it's a better solution than mine, as it looks fool proof against future changes in `hostinterfaces` fields in next Zabbix versions.

Reply:
Thanks this is a good point and I hadn't thought that through properly. I've corrected it, now the `update_interfaces = True` will compare all both lengths and the `matching_interfaces` to see if anything is wrong and needs to be updated.

Reply:
Great! I will do some tests to check if all is Ok as it seems to be.